command.go

```diff
@@ -477,6 +483,18 @@ func (c *Command) Help() error {
 	return nil
 }
 
+// SuggestFunc returns either the function set by SetSuggestFunc for this command
+// or a parent, or it returns a function with default suggestion behavior.
+func (c *Command) SuggestFunc() func(string) string {
+	if c.suggestFunc != nil && !c.DisableSuggestions {
+		return c.suggestFunc
+	}
+	if c.HasParent() {
+		return c.Parent().SuggestFunc()
+	}
+	return c.findSuggestions
+}
```

Comment:
Using this

```suggestion
func (c *Command) SuggestFunc(arg string) string {
	if c.suggestFunc != nil && !c.DisableSuggestions {
		return c.suggestFunc(arg)
	}
	if c.HasParent() {
		return c.Parent().SuggestFunc(arg)
	}
	return c.findSuggestions(arg)
}
```

Would avoid the strange way the function is called

```diff
-                                return fmt.Errorf("invalid argument %q for %q%s", v, cmd.CommandPath(), cmd.SuggestFunc()(args[0]))
+                                return fmt.Errorf("invalid argument %q for %q%s", v, cmd.CommandPath(), cmd.SuggestFunc(args[0]))
```